### PR TITLE
Let WS and HTTP use the same JSON encoder

### PR DESCRIPTION
**Pull request by DoctorJohn**
## Description

It was already possible and documented to override `encode_json` on views to customize the JSON encoder used for responses sent via HTTP. The same `encode_json` method is now also used to encode WebSocket messages.

This was proposed a few weeks ago by @bellini666 and it's super handy if someone wants to use `orjson` or otherwise customize JSON responses and messages.

Note that this change is kinda breaking since the `encode_json` methode previously only expected `GraphQLHTTPResponse` dicts, whereas now it must be able to handle any kind of json data.
In reality anyone overriding this method can probably already handle any json data.

(Will do the same for decoding in a separate PR).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Summary by Sourcery

Unify the JSON encoding process for HTTP and WebSocket by using the same encode_json method, allowing for consistent customization across both protocols. Update documentation and add tests to support this change.

New Features:
- Enable the use of a single JSON encoder for both HTTP responses and WebSocket messages by allowing the override of the encode_json method.

Documentation:
- Update documentation to reflect the change in the encode_json method, which now applies to both HTTP and WebSocket JSON responses.
- Add a breaking changes document for version 0.251.0, detailing the changes to the encode_json method signature.

Tests:
- Add a test to verify that WebSocket handlers use the view's encode_json method for encoding JSON messages.

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Starting with this release, the same JSON encoder is used to encode HTTP
responses and WebSocket messages.

This enables developers to override the `encode_json` method on their views to
customize the JSON encoder used by all web protocols.



Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to @NucleonJohn for the PR 👏

Get it here 👉 https://strawberry.rocks/release/(next)
```

File: docs/breaking-changes/0.251.0.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+```python
+def encode_json(self, response_data: GraphQLHTTPResponse) -> str: ...
+```
+
+To upgrade your code, change the method signature to the following and make sure
+your method can handle the same inputs as the built-in `json.dumps` method:
+
+```python
+def encode_json(self, data: object) -> str: ...
```
Comment (patrick91) [inline docs/breaking-changes/0.251.0.md:24]: could we type this better? is it worth it?

Comment (DoctorJohn) [reply docs/breaking-changes/0.251.0.md:24]: `json.loads` expects `Any` so I went with `object`. Technically it's anything that can be serialized to JSON. But the `json` library doesn't provide a type for it. We would have to create our own, not sure whether it's worth it.

Comment (patrick91) [reply docs/breaking-changes/0.251.0.md:24]: @DoctorJohn mmh, is it because the ws integration can send pretty much anything, right?

Comment (patrick91) [reply docs/breaking-changes/0.251.0.md:24]: ok, sounds good to me 😊

**Review by patrick91 — APPROVED** [review]

**Pull request merged**